src/base-element.js

```diff
@@ -802,6 +806,16 @@ export class BaseElement {
   }
 
   /**
+   * Called when an attribute's value changes.
+   * @param {!string} unusedName
+   * @param {?string} unusedOldValue
+   * @param {?string} unusedNewValue
+   */
+  attributeChangedCallback(unusedName, unusedOldValue, unusedNewValue) {
```

Comment:
So the notion of bulk change is lost in this API. E.g. the system might accumulate N changes for this element and we'll call them uncorrelated one at a time. How do you see an implementation would be able to reduce amount of work? Would we use vsync? E.g.

```
attributeChangedCallback(...) {
  switch (name) {
    case 'src':
       this.src_ = newValue;
       break;
    case 'abc':
       this.stateAbc_ = newValue;
       break;
  }
  this.getVsync().mutate(() => this.updateUi_);
}
```

Is this how you see it?